### DISCUSSION TITLE
Touch records

### DISCUSSION THREAD

**Discussion by ak-acieslik** [MarkMpn/Sql4Cds#581] [General]
Hi @mark-data8 ,
do you know if its possible to touch records in a table like you could do with the Bulk Data Updater?

Here is a short outline how the SQL code could look like:
``` sql
-- Create a temporary table to store the createdon values
CREATE TABLE #TempAccountCreatedOn (
    AccountId INT,
    CreatedOn DATETIME
);

-- Insert the createdon values from the account table into the temporary table
INSERT INTO #TempAccountCreatedOn (AccountId, CreatedOn)
SELECT AccountId, CreatedOn
FROM Account;

-- Update the createdon column in the account table with the values from the temporary table
UPDATE Account
SET CreatedOn = t.CreatedOn
FROM Account a
INNER JOIN #TempAccountCreatedOn t ON a.AccountId = t.AccountId;

-- Drop the temporary table as it is no longer needed
DROP TABLE #TempAccountCreatedOn;
```

Maybe you could implement something to simplify this use case or to provide a configuration to provide the filter criteria and the touch fields and then it would generate the final SQL statement based on that?

**Comment by MarkMpn** [comment]
You should be able to do this today without the temporary table, e.g.

```sql
UPDATE Account
SET CreatedOn = CreatedOn
WHERE ...
```

Does that do what you need?

**Reply by ak-acieslik** [reply]
Yes, thanks! Thats very helpful!
